### PR TITLE
public pages: better metas

### LINKED ISSUE #2990: Better meta for public profiles
"Sergio Álvarez Leiva is creating maps and visualizations using CartoDB" --> "Sergio Álvarez Leiva has created XX maps and published XX datasets in CartoDB" (As well as de FB and TW descriptions)

"Maps by Sergio Álvarez Leiva on CartoDB" --> "CartoDB · Sergio Álvarez Leiva"

Keywords: username, name, datasets, maps, data visualization, spatial data, geospatial, cartodb

### PR DESCRIPTION
**Pull request by matallo**
inferred from the issue description and extended the changes to "public tables" "public vizs" and "public user", the changes in the code are self explanatory 

btw, are public table views to receive new dashboard love? 

close #2990 

CR @xavijam @saleiva

### CODE REVIEW

File: app/views/admin/pages/new_public_datasets.html.erb [hunk 1]
```diff
@@ -1,11 +1,14 @@
-<%= content_for(:title) do %>
-  Datasets by <%= @name %> on CartoDB
-<% end %>
+<%= content_for(:title) do %>CartoDB · Datasets by <%= @name %><% end %>
 
-<%= content_for(:description) do %>
-  <%= @name %> is creating and sharing public datasets using CartoDB
+<!-- TODO: move to helper -->
```
Comment (xavijam) [inline app/views/admin/pages/new_public_datasets.html.erb]: Why don't you make this thing in the controller?

File: app/views/admin/pages/new_public_maps.html.erb [hunk 2]
```diff
@@ -1,11 +1,28 @@
-<%= content_for(:title) do %>
-  Maps by <%= @name %> on CartoDB
-<% end %>
+<%= content_for(:title) do %>CartoDB · Maps by <%= @name %><% end %>
+
+<!-- TODO: move to helper -->
+<% if @visualizations.size == 0 && @tables_num == 0 %>
```
Comment (xavijam) [inline app/views/admin/pages/new_public_maps.html.erb]: Same here

File: app/views/admin/pages/new_public_datasets.html.erb [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+<% if @tables_num == 0 %>
+  <% description = "#{@name} is creating and sharing public datasets in CartoDB" %>
```
Comment (saleiva) [inline app/views/admin/pages/new_public_datasets.html.erb]: Uhm... not sure about the text (Sorry if it was me proposing it).
What about "XXX has published XX public datasets in CartoDB · Open Data rocks!"

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3457/))

File: app/views/admin/pages/new_public_datasets.html.erb (hunk 1 above)
Comment (matallo) [inline app/views/admin/pages/new_public_datasets.html.erb]: good catch, as in `@description`?

File: app/views/layouts/application_public_visualization_layout.html.erb [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
   <head>
-    <meta name="viewport" content="initial-scale=1.0, user-scalable=no" />
-    <meta http-equiv="content-type" content="text/html; charset=UTF-8"/>
     <%= favicon_link_tag "/favicons/favicon.ico" %>
-    <title><%= @visualization.name.gsub(/_/," ") %> | CartoDB</title>
-    <meta name="description" content="<%= @visualization.description_clean || @visualization.name.gsub(/_/," ") %>">
-    <meta name="keywords" content="Map,cartodb,<%= @visualization.name.gsub(/_/,",").gsub(/ /,",") %>">
+    <title><%= yield :title %></title>
+    <meta name="description" content="<%= yield :description %>">
+    <meta name="keywords" content="<%= @visualization.name.gsub(/ /,", ").gsub(/_/,", ") %>, datasets, maps, data visualization, spatial data, geospatial, cartodb">
```
Comment (saleiva) [inline app/views/layouts/application_public_visualization_layout.html.erb]: ++ dataset, map, database, open data

File: app/views/admin/pages/new_public_datasets.html.erb (hunk 1 above)
Comment (xavijam) [inline app/views/admin/pages/new_public_datasets.html.erb]: yep

File: app/views/admin/pages/new_public_datasets.html.erb (hunk 3 above)
Comment (matallo) [inline app/views/admin/pages/new_public_datasets.html.erb]: okie dokie, in that direction I like how GitHub uses repo descriptions for user acquisition

screen shot 2015-05-05 at 11 49 09

what do you think?

**Comment by matallo** [comment]
moved description to controller and replaced texts, what about the description I pasted?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3465/))

**Comment by matallo** [comment]
bump, what about the description?

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3530/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3532/))

**Pull request merged**